debezium-core/src/main/java/io/debezium/jdbc/JdbcConfiguration.java

```diff
@@ -55,11 +62,25 @@
      */
     public static final Field ON_CONNECT_STATEMENTS = Field.create("initial.statements", "A semicolon separated list of statements to be executed on connection");
 
+    /**
+     * An optional field for datasource factory class that will be used to build the datasource connection pool.
+     */
+    public static final Field CONNECTION_FACTORY_CLASS = Field.create("connection.factory.class")
```

Comment:
We introduced how to plugin your own ConnnectorFactory class when you have different PooledDataSources. Ex: We introduced a HikariDataSource pool. 
If user specify any connection.factory.class it will honor it otherwise it will use default JdbConnection.ConnectionFactory.

Example implementation of ConnectionFactory
```java
public class HikariConnectionFactory implements JdbcConnection.ConnectionFactory {

/**
     * Create connection pool if not yet done and return a connection from pool
     * @param inConfig jdbc configuration
     * @return connection object from pool
     * @throws SQLException if any sql exception
     */
    @Override
    public Connection connect(JdbcConfiguration inConfig) throws SQLException {

        String vendor = inConfig.asProperties().getProperty("vendor");
        if (_dataSource == null) {
            String url;
            Properties props = inConfig.asProperties();
            HikariConfig hikariConfig = new HikariConfig();
            if ("oracle".equalsIgnoreCase(vendor)) {
                url = findAndReplace(ORACLE_URL_PATTERN, props);
                hikariConfig.setConnectionTestQuery("select 1 from dual");
            } else {
                if (inConfig.getInstance() != null &&
                        inConfig.getInstance().trim().length() > 0){
                    url = findAndReplace(SQLSERVER_INSTANCE_URL_PATTERN, props);
                } else {
                    url = findAndReplace(SQLSERVER_URL_PATTERN, props);
                }
            }
            LOGGER.info("jdbc connection url:" + url);
            hikariConfig.setJdbcUrl(url);
            hikariConfig.setUsername(inConfig.getUser());
            hikariConfig.setPassword(inConfig.getPassword());
            String maxPoolStr = inConfig.asProperties().getProperty("connection.max.poolsize");
            hikariConfig.setMaximumPoolSize(Integer.parseInt(maxPoolStr));
            hikariConfig.setConnectionTimeout(inConfig.getConnectionTimeoutMs());
            hikariConfig.setAutoCommit(false);
            hikariConfig.setPoolName("Debezium Hikari CP");
            hikariConfig.setRegisterMbeans(true);
            hikariConfig.addDataSourceProperty("cachePrepStmts",  true);
            hikariConfig.addDataSourceProperty("prepStmtCacheSize", 5);
            hikariConfig.addDataSourceProperty("prepStmtCacheSqlLimit", 50);
            hikariConfig.addDataSourceProperty("useServerPrepStmts", true);
            _dataSource = new HikariDataSource(hikariConfig);
        }
        return _dataSource.getConnection();
    }
} 

```